soroban-env-host/src/test/token.rs

```diff
@@ -501,6 +667,38 @@ fn test_classic_asset_roundtrip(asset_code: &[u8]) {
             .to_i64(),
         2
     );
+
+    // Make sure negative amounts are not allowed in smart/classic conversions.
+    assert!(token
+        .import(
+            &user,
+            token.nonce(user.get_identifier(&test.host)).unwrap(),
+            -1,
+        )
+        .is_err());
+    assert!(token
+        .export(
+            &user,
+            token.nonce(user.get_identifier(&test.host)).unwrap(),
+            i64::MIN,
+        )
+        .is_err());
+
+    // Also cover potential overflows
```

Comment:
👏🏻 The comments are very helpful in these tests.

Reply:
Thanks! I would we revisit the test framework though as these comments are a poor substitution for something like sections in catch2 in C++ (there is [rust-catch](https://github.com/guydunton/rust-catch), but not sure how stable it is).